vars/sbtReleaseDeployJob.groovy

```diff
@@ -0,0 +1,68 @@
+def call(Map config) {
+  library("tdr-jenkinslib")
+    
+  def versionBumpBranch = "version-bump-${config.buildNumber}"
+    
+  pipeline {
+    agent {
+      label "master"
+    }
+    //Parameters section required for display in client Jenkins jobs GUI.
+    //Selected values are not accessible within the function and must be passed into the function via the config map
+    parameters {
+      choice(name: "STAGE", choices: ["intg", "staging", "prod"], description: "The stage you are deploying the ${config.libraryName} library to")
+    }
+    stages {
```

Comment:
Do you need this parameters block if you're passing in the stage as the parameter map?

Reply:
Its for the client Jenkins job, to display the parameters if a manual build is required in the Jenkins GUI. For example deploying to staging/prod. See for example: https://jenkins.tdr-management.nationalarchives.gov.uk/job/TDR%20Auth%20Utils%20Deploy/

Reply:
Ok cool, makes sense.